### PR TITLE
NewXcodeVersions workflow polish work

### PR DESCRIPTION
**Pull request by isabelrios**
Trying to improve the way we exit the script so that the build is always green unless there is a real issue

### CODE REVIEW

File: bitrise.yml
Comment (isabelrios) [inline bitrise.yml:2981]: I need to undo this change before merging

File: bitrise.yml
Comment (isabelrios) [inline bitrise.yml:2959]: Not sure if adding the run_if here will work.
Let's comment this for now until we have the environment set as working, running unit tests would be enough.

**Comment by isabelrios** [comment]
See build working with the conditional statment: 
https://app.bitrise.io/build/92297bc007d38c47#?tab=log

File: bitrise.yml
Comment (AaronMT) [inline bitrise.yml:2779]: I think you need to wrap $resp in quotes as it will throw a ```[: too many arguments``` from https://stackoverflow.com/questions/13781216/meaning-of-too-many-arguments-error-from-if-square-brackets

```Bash
if [ -z "$resp" ]
```

File: bitrise.yml
Comment (AaronMT) [inline bitrise.yml:2784]: Does the value need quotations? Not familiar with ```envman``` key value pairs

**Review by AaronMT — CHANGES_REQUESTED** [review]

File: bitrise.yml
Comment (isabelrios) [reply bitrise.yml:2784]: I started using quotes but then I had to remove them to have the variable working for the run_if conditions in next steps, please see this post: https://discuss.bitrise.io/t/how-can-i-use-variables-stored-with-envman-in-the-run-if-check-before-a-step-is-run/3857/3

File: bitrise.yml
Comment (isabelrios) [reply bitrise.yml:2779]: This is working as it is, I tried both with empty resp (see [build](https://app.bitrise.io/build/2720b700e45762a9#?tab=log)) and with my name as committer (see [build](https://app.bitrise.io/build/2604ce2a8919202d#?tab=log), red due to running older xcode but steps running as defined by run_if) and both seemed to work...not sure I understand what would be the problem...

@AaronMT Let me know if that it is not what you mean please...

Comment (AaronMT) [reply bitrise.yml:2779]: It might depend on the environment. It's one of those things if it happens we'll fix it. 🤷🏼‍♂️

Comment (isabelrios) [reply bitrise.yml:2779]: I will modify this on Monday to wrap it as you suggests on Monday and we can land then along with the other PR

Comment (AaronMT) [reply bitrise.yml:2779]: Ok

**Pull request merged**
